### ISSUE TITLE
Exports are no longer recognized in js modules which contain functions with parameter called `module` which have `module.exports =` in their body

### ISSUE THREAD

**Issue by peetklecha** [microsoft/typescript-go#2656]
## Steps to reproduce

```ts
// dep.js
function something(module) {
    module.exports = 8;
}

export default 7;

// main.ts

import DefaultExport from "./dep.js";

export default 3 + DefaultExport;
```

## Behavior with `typescript@5.9`

No errors raised

## Behavior with `tsgo`

Raises TS1192 (or if `dep.js` does named exports and `main.ts` does named imports, TS2305).
